steps/24_categorical_support/step.md

```diff
@@ -0,0 +1,223 @@
+# Adding categorical support to sktime
+
+Contributors: @Abhay-Lejith
+
+## Introduction
+
+### Objectives 
+Allow users to pass exog data with categorical features to any forecaster. Outcome will depend on whether the forecaster natively supports categorical or not.
+Two broadly possible outcomes:
+- error in fit saying not natively supported.
+- internally encode and fit successfully.
+
+### Current status:
+* Non-numeric dtypes are completely blocked off in datatype checks.
+* https://github.com/sktime/sktime/pull/5886 - This pr removes these checks but has not been merged yet as most estimators do not support categorical and will break without informative error messages.
+* https://github.com/sktime/sktime/pull/6490 - pr adding feature_kind metadata
+
+
+### Models which support categorical natively that have been identified so far:
+- xgboost
```

Comment:
how does this compare to the list in
https://github.com/sktime/sktime/issues/6109#issuecomment-1989464887
and subsequent discussion?

Reply:
This list is of forecasters with native support. And I am currently looking at only categorical in exog.